### PR TITLE
changes for optimzing the show command

### PR DESCRIPTION
**Pull request by syed-awais-ali**
@edx/devops Review please.

### CODE REVIEW

**Comment by jibsheet** [comment]
FYI @doctoryes since you filed the original ticket

File: plugins/show.py [hunk 1]
```diff
@@ -419,17 +420,15 @@ def _show_edp(self, message, env, dep, play):
...
-                    """
-                    Return name of the instance.
-                    """
-                    return instance.name
-                elbs = map(instance_name,
-                           self._instance_elbs(instance.id, dep))
+                elb_list = []
+                for elb in elbs:
+                    lb_instance_ids = [inst.id for inst in elb.instances]
+                    if instance.id in lb_instance_ids:
```
Comment (jibsheet) [inline plugins/show.py]: I assume there's some clever python generator thing we could do here, but elb.instances should be already fetched from the API.  Traces in the ticket already show a huge API win.

Comment (doctoryes) [inline plugins/show.py]: Efficient and easy-to-read always beats clever in my book... 😺

**Comment by syed-awais-ali** [comment]
@doctoryes here is the link to the ticket. https://openedx.atlassian.net/browse/OPS-2216. I have pasted the the comparison of the code wrt to API calls and time taken on my PC.

**Comment by doctoryes** [comment]
Thanks, @syed-awais-ali . I saw the outputs of both old and new code in the ticket - were the outputs equivalent? The reduction in boto calls is great.

**Comment by syed-awais-ali** [comment]
@doctoryes Yes, I checked the output for stage and that seems to be matching

**Comment by doctoryes** [comment]
Then merge away! I assume you have the needed permissions?

**Pull request merged**